### ISSUE TITLE
People: When removing a user, content attribution does not work

### ISSUE THREAD

**Issue by supernovia** [Automattic/wp-calypso#32079]
#### Steps to reproduce
1. Starting at https://wordpress.com/people, select a site that has multiple users
2. Click on a user that has at least one published post (perhaps even open that post in a new tab)
3. Choose "Attribute all content to" and select another user, then click "Delete User"

#### What I expected
The removed user should no longer be listed as the post author

#### What happened instead
They are still listed as the post author

#### Browser / OS version
Chrome / Mac (both current)

#### Screenshot / Video
![Screen Shot 2019-04-05 at 2 38 01 PM](https://user-images.githubusercontent.com/3476361/55655141-8fcdca80-57b0-11e9-9d10-292774a7430a.png)

but this is what I still see:

![Screen Shot 2019-04-05 at 2 23 57 PM](https://user-images.githubusercontent.com/3476361/55655149-99efc900-57b0-11e9-8f5b-94a4e0e6a15d.png)

(noting I tried clearing cache, too, and confirmed the post still shows up as owned by that user in back-end listings too)

#### Context / Source
Complaint in the forums:
https://en.forums.wordpress.com/topic/deleted-user-still-associated-with-posts/

Tested on a test site. :)

**Comment by arunsathiya** [comment]
@supernovia, I tested on a Jetpack site and it appears the content attribution works just fine. Here is an example of how things worked.

At the start of the test, I had 101 posts on the site. Posts 91 to 101 were assigned to `arunsathiyacontributor`. You can see so on [these images](https://cloudup.com/c_lKEyefw0D). On deleting this user, I reassigned these posts to `arunsathiyaauthor` and that seems to have worked as per the images [here](https://cloudup.com/cK32YENF9vE). 

But, on a WordPress.com site, the posts were not reassigned properly. At the start of the post, there were two posts with `arunsathiyacontributor` as shown [here](https://cloudup.com/c1EocFbTqpG), but on reassigning to `arunsathiyaauthor`, it did not work, as shown [here](https://cloudup.com/cLVRGpkhK7B).

Could you check if Jetpack sites work fine for you? Thanks!

**Comment by supernovia** [comment]
Just noting this isn't stale; I've confirmed again that when using 
https://wordpress.com/people/team/
to remove a user, I'm offered the chance to attribute all posts to another user on a site. 

I've removed a user, and attributed their posts to my account, but the posts still show up as being under the removed user.

**Comment by annemirasol** [comment]
Confirming that this still happens for WordPress.com sites.

When removing a user, admin is offered the option to reassign or delete posts, neither of which happens.

**Comment by flootr** [comment]
I'll have a look at this as I'm working on this piece of Calypso currently

**Comment by annemirasol** [comment]
> I'll have a look at this as I'm working on this piece of Calypso currently

Thanks for taking this on!

A somewhat related issue, the reason I stumbled onto this issue -- 2221-gh-Automattic/p2 
Essentially, some users are asking if reassigning or deleting content can be made purely optional when removing a user from a blog.

**Comment by supernovia** [comment]
> Could you check if Jetpack sites work fine for you? Thanks!

Ack! Sorry I missed this. I have confirmed it's fine on my Jetpack on .org site.

**Comment by supernovia** [comment]
As an update:

- Still works in Jetpack (at least when testing with wp-admin... 🤔  I should actually try Jetpack Calypso..)

- I tested again and content is still not properly attributed in calypso. The tool says that I have the choice of deleting all of the user's content, or attributing it to someone else. I chose to attribute. The box also looks wonky now with a sideways scroll instead of a dropdown:
<img width="757" alt="Screen Shot 2021-10-28 at 11 28 15 AM" src="https://user-images.githubusercontent.com/3476361/139306154-39a7e8c7-6309-4204-ab78-a43e99fd8970.png">

I selected another user anyway
<img width="749" alt="Screen Shot 2021-10-28 at 11 28 25 AM" src="https://user-images.githubusercontent.com/3476361/139306256-3dc05d72-d242-464f-bc99-994f3684dc82.png">

I deleted her successfully.

Then I checked my site again; the post is still attributed to astrofishassist.
<img width="1058" alt="Screen Shot 2021-10-28 at 11 28 57 AM" src="https://user-images.githubusercontent.com/3476361/139306425-20c5d8ed-6f10-40dd-bf54-e951061f54ac.png">
